diff --git a/meson.build b/meson.build
index 6fd12dc..dad3844 100644
--- a/meson.build
+++ b/meson.build
@@ -319,12 +319,7 @@ else
 endif
 
 # BZip2 support.
-bzip2_dep = dependency('bzip2', required: false)
-if not bzip2_dep.found()
-  bzip2_dep = cc.find_library('bz2',
-    has_headers: ['bzlib.h'],
-    required: get_option('bzip2'))
-endif
+bzip2_dep = disabler()
 
 if bzip2_dep.found()
   ftoption_command += ['--enable=FT_CONFIG_OPTION_USE_BZIP2']
