diff --git a/meson.build b/meson.build
index 53ba3d3..be24a31 100644
--- a/meson.build
+++ b/meson.build
@@ -459,6 +459,11 @@ python_env.append('PYTHONPATH', thirdparty/'glad')
 
 if host_machine.system() == 'windows'
-    threads = declare_dependency()
+    threads = declare_dependency(
+      dependencies: [
+        cc.find_library('winpthread', static: true),
+        cxx.find_library('stdc++', static: true),
+      ],
+    )
 else
     pthreads = dependency('threads')
     has_setclock = cc.has_header_symbol(

